drivers/bluetooth/hci/ipm_stm32wb.c

```diff
@@ -327,17 +327,37 @@ static void start_ble_rf(void)
 		LL_RCC_ReleaseBackupDomainReset();
 	}
 
-	/* Select LSE clock */
-	LL_RCC_LSE_Enable();
-	while (!LL_RCC_LSE_IsReady()) {
-	}
+	do {
+#ifdef CONFIG_CLOCK_STM32_LSE
+		const u32_t alarm = k_uptime_get_32() + K_MSEC(RCC_LSE_TIMEOUT_VALUE);
+		/* Select LSE clock */
+		LL_RCC_LSE_Enable();
+		while (!LL_RCC_LSE_IsReady()) {
+			if (alarm < k_uptime_get_32())
+				goto init_LSI;
+		}
+		/* Select wakeup source of BLE RF */
+		LL_RCC_SetRFWKPClockSource(LL_RCC_RFWKP_CLKSOURCE_LSE);
+		LL_RCC_SetRTCClockSource(LL_RCC_RTC_CLKSOURCE_LSE);
+
+		/* Switch OFF LSI */
+		LL_RCC_LSI1_Disable();
+		break;
+#endif
+init_LSI:
+		__HAL_RCC_LSI1_ENABLE();
```

Comment:
I need to cross check that RTC is actually required.
Anyway a comment indicating the match LSI1>RTC, LSI2>RF would be nice

Reply:
Wasn't sure about the actual LSI configuration. Datasheet claims that only LSI2 is required for RF, but it's better to be sure. I will try without LSI1 and see how it goes.

Reply:
without LSI1 works fine. I've removed it. Comment added

Reply:
Thanks, you can remove the LSI1 comment then ;-).
Also, I think we can actually remove the RTC stuff.
Could be a dedicated commit since this is rather a clean up.

Reply:
Make sense. Updated usage of LSI, replaced LSI1 -> LSI2